### ISSUE TITLE
Json serialization of binary data

### ISSUE THREAD

**Issue by jrmbeck** [Microsoft/bond#441]
Hi,

I'm trying to build a bond model that would be used to serialize either in Binary Bond or in Json.

In my model, I have a byte array (blob in bond), and a Guid (also blob, but aliased as a Guid in C#).
When I serialize the model in Json, both are serialized as an array of byte values (like [1, 24, 53]). I would expect the Guid to have the classical string representation, especially since the C# generated class has the Guid type, and the byte array to be serialized as a base64 string.
Is there a way to do this with the SimpleJsonReader/Writer?

I was able to obtain the expected behavior using directly Newtonsoft Json and some custom type serialization, but I would like to use the bond library, in particular to benefit from the constraint checks (required values).

Thanks,

  Jerome BECK

**Comment by chwarr** [comment]
In this case, you'll have to pick between serialized representations for GUID. If you want model it with the underlying type of blob, then it will be serialized as a blob in all the serialization protocols, SimpleJson included. (And, since there's no explicit blob type at the level of serialization, it gets turned into a list of uint8 values.) If you'd rather optimize for human-readability, you can use a string as the underlying type. The [guid example](https://github.com/Microsoft/bond/tree/master/examples/cs/core/guid) demonstrates representing a GUID as both a blob and as a string.

In other words, they're no way to have a field have a different underlying type per protocol when using type aliases. (For example, such a feature would make transcoding between protocols difficult, as the schema of the types being transcoded would be required, as would their custom converters.)

**Comment by chwarr** [comment]
Closing for now. Please re-open if needed.
